qiskit/circuit/qpy_serialization.py

```diff
@@ -981,15 +1070,22 @@ def _read_instruction(file_obj, circuit, registers, custom_instructions, version
         gate_class = getattr(extensions, gate_name)
     elif hasattr(quantum_initializer, gate_name):
         gate_class = getattr(quantum_initializer, gate_name)
+    elif hasattr(controlflow, gate_name):
+        gate_class = getattr(controlflow, gate_name)
     else:
         raise AttributeError("Invalid instruction type: %s" % gate_name)
-    if gate_name == "Initialize":
-        gate = gate_class(params)
+    if gate_name in {"IfElseOp", "WhileLoopOp"}:
+        gate = gate_class(condition_tuple, *params)
     else:
-        if gate_name == "Barrier":
-            params = [len(qargs)]
-        gate = gate_class(*params)
-    gate.condition = condition_tuple
+        if gate_name == "Initialize":
+            gate = gate_class(params)
+        else:
+            if gate_name == "Barrier":
+                params = [len(qargs)]
+            elif gate_name == "BreakLoopOp":
+                params = [len(qargs), len(cargs)]
```

Comment:
`ContinueLoopOp` will be the same as `BreakLoopOp` - in tests of control flow, I almost invariably just test both of them in the same tests because they're basically identical.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7584/commits/788f9432dba864f193cef1c23fcff84eed58ee21